docs/docs/visual-testing-with-storybook.md

```diff
@@ -21,97 +21,200 @@ npx -p @storybook/cli sb init
+// __PATH_PREFIX__ is used inside gatsby-link an other various places. For storybook not to crash we need to set it as well.
 global.__PATH_PREFIX__ = ""
 
-// This is to utilized to override the window.___navigate method Gatsby defines and uses to report what path a Link would be taking us to if it wasn't inside a storybook
+// Navigating through a gatsby app using gatsby-link or any other gatsby component will use the `___navigate` method.
+// In storybook it doesn't make sense to do an actual navigate, instead we want to log an action. Checkout the actions addon docs https://github.com/storybookjs/storybook/tree/master/addons/actions.
+
 window.___navigate = pathname => {
   action("NavigateTo:")(pathname)
 }
-// highlight-end
 ```
 
-> You can remove the `stories` folder from the root of your project, or move it inside your `src` folder
-
-Next make some adjustments to Storybook's default `webpack` configuration so you can transpile Gatsby source files, and to ensure you have the necessary `babel` plugins to transpile Gatsby components.
+### Add TypeScript Support
 
-Create a new file called `webpack.config.js` in the `.storybook` folder created by the Storybook CLI. Then place the following code in that file (depending on which version of Storybook you're using):
+To configure TypeScript with Storybook and Gatsby, add the following configuration to `.storybook/main.js`.
 
-**For Storybook v5:**
+Add `tsx` as a file type to look for in the `stories` array. This is assuming the default Storybook path, but this can be modified to where your stories live.
 
-```js:title=.storybook/webpack.config.js
-module.exports = ({ config }) => {
-  // Transpile Gatsby module because Gatsby includes un-transpiled ES6 code.
-  config.module.rules[0].exclude = [/node_modules\/(?!(gatsby)\/)/]
-
-  // use installed babel-loader which is v8.0-beta (which is meant to work with @babel/core@7)
-  config.module.rules[0].use[0].loader = require.resolve("babel-loader")
-
-  // use @babel/preset-react for JSX and env (instead of staged presets)
-  config.module.rules[0].use[0].options.presets = [
-    require.resolve("@babel/preset-react"),
-    require.resolve("@babel/preset-env"),
-  ]
-
-  config.module.rules[0].use[0].options.plugins = [
-    // use @babel/plugin-proposal-class-properties for class arrow functions
-    require.resolve("@babel/plugin-proposal-class-properties"),
-    // use babel-plugin-remove-graphql-queries to remove static queries from components when rendering in storybook
-    require.resolve("babel-plugin-remove-graphql-queries"),
-  ]
+```js:title=.storybook/main.js
+stories: ["../stories/**/*.stories.js","../stories/**/*.stories.tsx"],
+```
 
-  // Prefer Gatsby ES6 entrypoint (module) over commonjs (main) entrypoint
-  config.resolve.mainFields = ["browser", "module", "main"]
+After line 25 containing `config.resolve.mainFields = ["browser", "module", "main"];` and before line 27 containing `return config;`
```

Comment:
To make sure we don't introduce docs debt or confusion: is there a more general way to describe the job to be done in case the user has existing info in their Storybook configuration (i.e. can you count on line numbers here)?

Reply:
Hi @marcysutton - My thought process is the user could compare this to a fresh install of `main.js` to figure out where to put the lines if their configuration differed. I did this because when I went through this initially, I kept thinking, "Where am I supposed to put this?!" Using line numbers was the only way I could think of to accurately explain where to put the new configuration, but I'm open to changing this if you feel it will cause issues going forward.

Reply:
That makes sense–I'm mostly wondering if counting on line numbers from a fresh install would be brittle, since that file is generated by Storybook. If you describe it more generally _and_ hint with line numbers, would it still work?

```suggestion
Add the following code after the line containing `config.resolve.mainFields = ["browser", "module", "main"];` (line 25) and before the `return config;` in the same function body (line 27):
```